### PR TITLE
QE: Fix variables for GitHub and container runtime

### PR DESCRIPTION
**Pull request by nodeg**
## What does this PR change?

This PR
- renames the 2 variables used for the GitHub validation and the container runtime
- uses the correct one in the ISO mounting steps that failed in the 5.0 BV because of that before

See https://github.com/SUSE/spacewalk/issues/23705

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes
- [x] **DONE**

## Test coverage
- Cucumber tests were edited
- [x] **DONE**

## Links

Issue(s): #
Ports(s): # No port needed 

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Review by Bischoff — COMMENTED** [review]
what is a "container runtime" ?

is it the same as a "containerized server" ? if yes, I find the second clearer.

**Comment by nodeg** [comment]
> what is a "container runtime" ?

`$has_container_runtime` means that we are on a containerized system and use e.g. the `podman` runtime
https://github.com/SUSE/susemanager-ci/blob/eff680b4bd7e6f0422af1032d126a89a236c05f3/terracumber_config/tf_files/SUSEManager-5.0-build-validation-NUE.tf#L179

> is it the same as a "containerized server" ? if yes, I find the second clearer.

Not necessary server, but also e.g. proxy.

**Comment by Bischoff** [comment]
> > what is a "container runtime" ?
> 
> `$has_container_runtime` means that we are on a containerized system and use e.g. the `podman` runtime https://github.com/SUSE/susemanager-ci/blob/eff680b4bd7e6f0422af1032d126a89a236c05f3/terracumber_config/tf_files/SUSEManager-5.0-build-validation-NUE.tf#L179
> 
> > is it the same as a "containerized server" ? if yes, I find the second clearer.
> 
> Not necessary server, but also e.g. proxy.

Use `containerized_system` then ?

Just a suggestion. Feel free to ignore.

File: testsuite/features/support/env.rb [hunk 1]
```diff
@@ -567,19 +567,19 @@ def process_code_coverage
   skip_this_scenario if $is_cloud_provider
 end
 
-# skip tests if executed in containers for the githug validation
+# skip tests if executed in containers for the GitHub validation
```
Comment (Bischoff) [inline testsuite/features/support/env.rb:570]: LOL

**Review by srbarrios — APPROVED** [review]

**Comment by srbarrios** [comment]
> what is a "container runtime" ?
> 
> is it the same as a "containerized server" ? if yes, I find the second clearer.

https://opensource.com/article/21/9/container-runtimes

**Review by Bischoff — CHANGES_REQUESTED** [review]
The current tags are
```
@skip_if_container_server
@skip_if_not_container_server
```

1) the global variable name should match the tag
2) "has container runtime" is really unclear
3) "skip_if_not_" does not match our conventions and should be replaced with ""

I suggest:
```
$is_containerized_system     (or server)
@skip_if_containerized_system
@containerized_system
```

**Comment by Bischoff** [comment]
> > what is a "container runtime" ?
> > is it the same as a "containerized server" ? if yes, I find the second clearer.
> 
> https://opensource.com/article/21/9/container-runtimes

I agree, but there's no point in adding more technical mambo-jambo.

Everyone in the team is so far speaking of the containerized server. Let's stick to it.

**Review by srbarrios — APPROVED** [review]

**Review by Bischoff — APPROVED** [review]
Wonderful. Thanks so much.

File: testsuite/features/support/env.rb [hunk 2]
```diff
@@ -53,8 +53,8 @@
 Capybara.default_max_wait_time = ENV['CAPYBARA_TIMEOUT'] ? ENV['CAPYBARA_TIMEOUT'].to_i : 10
 DEFAULT_TIMEOUT = ENV['DEFAULT_TIMEOUT'] ? ENV['DEFAULT_TIMEOUT'].to_i : 250
 $is_cloud_provider = ENV['PROVIDER'].include? 'aws'
-$is_container_provider = ENV['PROVIDER'].include? 'podman'
-$is_container_server = %w[k3s podman].include? ENV.fetch('CONTAINER_RUNTIME', '')
+$is_gh_validation = ENV['PROVIDER'].include? 'podman'
```
Comment (NamelessOne91) [inline testsuite/features/support/env.rb:56]: Just a doubt/nitpick

I see the env variable being set in the bash script under the  podman_runner directory, but why do we check for the PROVIDER to be  `podman` to know if we're running tests on GitHub ?
If it is an arbitrary value, maybe we can mention GH in it.

**Review by jordimassaguerpla — APPROVED** [review]

File: testsuite/features/support/env.rb (hunk 2 above)
Comment (nodeg) [reply testsuite/features/support/env.rb:56]: Good point, Paolo. This needs to be clarified with @jordimassaguerpla, and we can improve this in a follow-up PR :wink:

**Pull request merged**